src/network/receiver.go

```diff
@@ -0,0 +1,57 @@
+package network
+
+import (
+	"net"
+	"strconv"
+	"time"
+)
+
+/*
+Listen for incoming messages on specified IP and port.
+*/
+func ListenForMessages(ip string, port int, messageChannelchan chan<- []byte) {
+	conn, err := net.ListenPacket("udp4", ip+":"+strconv.Itoa(port))
+
+	if err != nil {
+		messageChannelchan <- []byte("CONNECTION ERROR")
```

Comment:
I think we should change this to a _panic_ for now. We will create proper error handling later, but we need to discuss how to do it first.